### DISCUSSION TITLE
Classification of Kits to Maturity Levels

### DISCUSSION THREAD

**Discussion by arnoweiss** [eclipse-tractusx/eclipse-tractusx.github.io#881] [General]
For a few weeks, there's been a released [graduation process for kits](https://eclipse-tractusx.github.io/documentation/kit-maturity-levels). That's good because it provides quick feedback cycles and clear obligations. From a governance perspective, it raises a couple of questions:
1. How to deal with existing Kits? There are [several](https://eclipse-tractusx.github.io/docs-kits/kits/Data%20Governance%20Kit/Data%20Governance%20Kit%20Adoption%20View) [Kits](https://eclipse-tractusx.github.io/docs-kits/kits/ESS-Kit/ESS%20Kit%20Adoption%20View) [that](https://eclipse-tractusx.github.io/docs-kits/kits/Behaviour%20Twin%20HI%20Kit/Adoption%20View%20Health%20Indicator%20Kit) [wouldn't](https://eclipse-tractusx.github.io/docs-kits/kits/Behaviour%20Twin%20MDP%20Kit/Adoption%20View%20Model%20Based%20Development%20and%20Data%20Processing%20Kit) [make it](https://eclipse-tractusx.github.io/docs-kits/kits/Manufacturing%20as%20a%20Service%20Kit/Adoption%20View%20MaaS%20KIT) past "Sandbox". And there I'm not even talking of those that checked the boxes but aren't [nearly](https://eclipse-tractusx.github.io/docs-kits/kits/Modular%20Production%20Kit/Adoption%20View%20Modular%20Production%20Kit) [specific](https://eclipse-tractusx.github.io/docs-kits/next/kits/OSim%20Kit/Adoption%20View%20OSim%20Kit) enough for anyone to build a dataspace use-case.
2. As a reader, how do I recognize the current state a Kit is in? At least, I would expect that all "Sandbox"-Kits are in the "Upcoming Kits" section. But currently, they're all listed side-by-side.
3. What happens when a Kit doesn't make it past a certain stage? Removal from the release would be the logical step, I think. But the unhappy path isn't specified.
4. The [Graduation Process](https://eclipse-tractusx.github.io/documentation/kit-getting-started) references TRGs 9.xx that [aren't published](https://eclipse-tractusx.github.io/docs/release) and I couldn't find PRs for. What am I missing?

I'd love to hear some thoughts from the community. This discussion may result in some TRG drafts.

**Comment by maximilianong** [comment]
Hi @arnoweiss ,

@danielmiehle and me started with an assessment: 

1. 
KIT | PILE
-- | --
Agents | Graduated
Behaviour Twin | Graduated
Business Partner | Graduated +
Circularity | Incubating
Connector | Graduated
Data Chain | Graduated
DCM | Incubating
Data Governance | Sandbox
Dgital Twin | Graduated
Eco Pass | Graduated
ESS | Sandbox
Industry Core | Graduated
Model Based Development | Incubating
Modular Production | Incubating
Manufacturing as a Service | Incubating
OSim | Incubating
PURIS | Incubating
PCF | Incubating
Quality | Incubating
Traceabiity | Graduated

https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/issues/917

2. we dont make a category like "upcoming", want to make the maturity level transparent in two ways. One with a filter on the KITs page and on the other side with graphical dots: https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/issues/917

3. we do not expect that with every release the maturity level increases but yes, you are right, maybe we should track when a KIT stays for 3 releases in 'Sandbox' it gets deprecated or something like this.

4. I raised this one: https://eclipse-tractusx.github.io/docs/release/trg-0/trg-9-01 but it never got out of the drafts section.

**Reply by arnoweiss** [reply]
Thanks for compiling this. I'd prefer if the stages were stricter but given the rules, the levels seem consistent.

**Reply by maximilianong** [reply]
Can be extended :D This was basically the beginning and maybe we need to extend the process from stage to stage with the next release.

**Reply by arnoweiss** [reply]
My perspective is that no kit should graduate without a Reference Implementation

**Comment by stephanbcbauer** [comment]
@jSchuetz88

**Comment by matbmoser** [comment]
I have an updated list:

Kit Name | Current Level | Graduation Potential
-- | -- | --
behaviour-twin-kit | Incubating | 🟡 Medium
business-partner-kit | Graduated* | 🟢 High
circularity-kit | Incubating | 🟢 High
connector-kit | Graduated* | 🟢 High
customs-kit | Sandbox | Low
data-chain-kit | Graduated* | 🟢 High
data-driven-quality-management-kit | Incubating | 🟢 High
data-governance-kit | Sandbox | Low
data-trust-and-security-kit | Incubating | 🟢 High
demand-and-capacity-management-kit | Graduated* | -
digital-twin-kit | Graduated* | -
eco-pass-kit | Incubating | 🟢 High
environmental-and-social-standards-kit | Sandbox | Low
industry-core-kit | Graduated* | -
knowledge-agents-kit | Incubating | Low
logistics-kit | Sandbox | Low
manufacturing-as-a-service-kit | Sandbox | Low
model-based-development-and-data-processing-kit | Sandbox | Low
modular-production-kit | Incubating | 🟡 Medium
online-simulation-kit | Sandbox | Low
product-carbon-footprint-exchange-kit | Incubating | 🟢 High
puris-kit | Graduated* | -
supply-chain-disruption-notification-kit | Graduated* | -
traceability-kit | Graduated* | -
requirements-kit | Sandbox | -

**Comment by matbmoser** [comment]
I will paste this in to the website
